distributed/tests/test_s3.py

```diff
@@ -27,60 +27,154 @@
-def test_read_bytes(e, s, a, b):
-    futures = read_bytes(test_bucket_name, prefix='test/', anon=True,
-                         lazy=False)
+    assert (s3.cat('/'.join([test_bucket_name, k])) ==
+            s3.cat('s3://' + '/'.join([test_bucket_name, k])))
+
+
+def test_seek_delimiter(s3):
+    fn = 'test/accounts.1.json'
+    data = files[fn]
+    with s3.open('/'.join([test_bucket_name, fn])) as f:
+        seek_delimiter(f, b'}', 0)
+        assert f.tell() == 0
+        f.seek(1)
+        seek_delimiter(f, b'}', 5)
+        assert f.tell() == data.index(b'}') + 1
+        seek_delimiter(f, b'\n', 5)
+        assert f.tell() == data.index(b'\n') + 1
+
+
+def test_read_block(s3):
+    import io
+    data = files['test/accounts.1.json']
+    lines = io.BytesIO(data).readlines()
+    path = 'distributed-test/test/accounts.1.json'
+    assert s3.read_block(path, 1, 35, b'\n') == lines[1]
+    assert s3.read_block(path, 0, 30, b'\n') == lines[0]
+    assert s3.read_block(path, 0, 35, b'\n') == lines[0] + lines[1]
+    assert s3.read_block(path, 0, 5000, b'\n') == data
+    assert len(s3.read_block(path, 0, 5)) == 5
+    assert len(s3.read_block(path, 0, 5000)) == len(data)
+    assert s3.read_block(path, 5000, 5010) == b''
+
+
+@gen_cluster(timeout=60)
+def test_read_bytes(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
+
+    futures = read_bytes(test_bucket_name+'/test/accounts.*', lazy=False)
     assert len(futures) >= len(files)
     results = yield e._gather(futures)
-    assert set(results).issuperset(set(files.values()))
-
-
-@gen_cluster(timeout=60, executor=True)
-def test_read_bytes_lazy(e, s, a, b):
-    values = read_bytes(test_bucket_name, 'test/', lazy=True, anon=True)
+    assert set(results) == set(files.values())
+
+    yield e._shutdown()
+
+
+@gen_cluster(timeout=60)
+def test_read_bytes_block(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
+    for bs in [5, 15, 45, 1500]:
+        vals = read_bytes(test_bucket_name+'/test/account*', blocksize=bs)
+        assert len(vals) == sum([(len(v) // bs + 1) for v in files.values()])
+        futures = e.compute(vals)
+        results = yield e._gather(futures)
+        assert sum(len(r) for r in results) == sum(len(v) for v in
+                   files.values())
+        futures = read_bytes(test_bucket_name+'/test/accounts*', blocksize=bs,
+                             lazy=False)
+        assert len(vals) == len(futures)
+        results = yield e._gather(futures)
+        assert sum(len(r) for r in results) == sum(len(v) for v in
+                   files.values())
+        ourlines = b"".join(results).split(b'\n')
+        testlines = b"".join(files.values()).split(b'\n')
+        assert set(ourlines) == set(testlines)
+    yield e._shutdown()
+
+
+@gen_cluster(timeout=60)
+def test_read_bytes_delimited(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield e._start()
```

Comment:
If you're interested, I've added an `executor=True` keyword argument to `gen_cluster` to automate the construction and shutdown of the exectuor:

```python
@gen_cluster(timeout=60, executor=True)
def test_read_bytes_delimited(e, s, a, b):
   ...
```